core/src/test/java/org/apache/calcite/test/ReflectiveSchemaTest.java

```diff
@@ -788,22 +788,23 @@ private static boolean isNumeric(Class type) {
 
   /** Test case for
    * <a href="https://issues.apache.org/jira/browse/CALCITE-1569">[CALCITE-1569]
-   * Date condition can generates Integer == Integer, which is always
-   * false</a>. */
+   * Date condition can generates Integer == Integer, which is always false</a> and
+   * <a href="https://issues.apache.org/jira/browse/CALCITE-3512">[CALCITE-3512]
+   * Query fails when comparing Time/TimeStamp types</a>. */
   @Test public void testDateCanCompare() {
-    final String sql = "select a.v\n"
-        + "from (select \"sqlDate\" v\n"
+    final String sql = "select a.v1, a.v2, a.v3\n"
```

Comment:
Oops, this test is messed up, i think we better split it into 3 tests: `testDateCanCompare` `testTimeCanCompare` and `testTimestampCanCompare`

Reply:
Addressed, thanks!